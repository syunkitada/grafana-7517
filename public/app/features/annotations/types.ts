import { PanelData, DataFrame, AnnotationEvent, TimeRange } from '@grafana/data';
import { DashboardModel, PanelModel } from '../dashboard/state';

export interface AnnotationQueryOptions {
  dashboard: DashboardModel;
  panel: PanelModel;
  range: TimeRange;
}

export interface AnnotationQueryResponse {
  /**
   * All the data flattened to a single frame
   */
  frame?: DataFrame;

  /**
   * The processed annotation events
   */
  events?: AnnotationEvent[];

  /**
   * The original panel response
   */
  panelData?: PanelData;
}
