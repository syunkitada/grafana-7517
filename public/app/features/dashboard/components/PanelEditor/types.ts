import { PanelModel } from '../../state/PanelModel';

export enum DisplayMode {
  Fill = 0,
  Fit = 1,
  Exact = 2,
}

export const displayModes = [
  { value: DisplayMode.Fill, label: 'Fill', description: 'Use all avaliable space' },
  { value: DisplayMode.Fit, label: 'Fit', description: 'Fit in the space keeping ratio' },
  { value: DisplayMode.Exact, label: 'Exact', description: 'Same size as the dashboard' },
];

export enum EditorTab {
  Query = 'query',
  Alerts = 'alerts',
  Transform = 'xform',
}

export const allTabs = [
  { tab: EditorTab.Query, label: 'Query', show: (panel: PanelModel) => true },
  { tab: EditorTab.Alerts, label: 'Alerts', show: (panel: PanelModel) => true },
  { tab: EditorTab.Transform, label: 'Transform', show: (panel: PanelModel) => true },
];
