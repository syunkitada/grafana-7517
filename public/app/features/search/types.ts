export enum DashboardSearchItemType {
  DashDB = 'dash-db',
  DashHome = 'dash-home',
  DashFolder = 'dash-folder',
}

export interface DashboardSection {
  id: number;
  uid?: string;
  title: string;
  expanded?: boolean;
  url: string;
  icon: string;
  score: number;
  hideHeader?: boolean;
  checked?: boolean;
  items: DashboardSectionItem[];
  toggle?: (section: DashboardSection) => Promise<DashboardSection>;
  selected?: boolean;
  type: DashboardSearchItemType;
}

export interface DashboardSectionItem {
  checked?: boolean;
  folderId?: number;
  folderTitle?: string;
  folderUid?: string;
  folderUrl?: string;
  id: number;
  isStarred: boolean;
  selected?: boolean;
  tags: string[];
  title: string;
  type: DashboardSearchItemType;
  uid: string;
  uri: string;
  url: string;
}

export interface DashboardTag {
  term: string;
  count: number;
}

export interface DashboardQuery {
  query: string;
  mode: string;
  tag: string[];
  starred: boolean;
  skipRecent: boolean;
  skipStarred: boolean;
  folderIds: number[];
}

export interface SectionsState {
  sections: DashboardSection[];
  allChecked: boolean;
  dashboardTags: DashboardTag[];
}

export type ItemClickWithEvent = (item: DashboardSectionItem | DashboardSection, event: any) => void;

export type SearchAction = {
  type: string;
  payload?: any;
};

export interface OpenSearchParams {
  query?: string;
}
