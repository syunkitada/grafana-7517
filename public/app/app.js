define([
  'angular',
  'jquery',
  'lodash',
  'app/core/config',
  'require',
  'bootstrap',
  'angular-route',
  'angular-sanitize',
  'angular-strap',
  'angular-dragdrop',
  'angular-ui',
  'bindonce',
  'app/core/core',
],
function (angular, $, _, config, appLevelRequire) {
  "use strict";

  var app = angular.module('grafana', []);
  var register_fns = {};
  var preBootModules = [];

  // This stores the grafana version number
  app.constant('grafanaVersion',"@grafanaVersion@");

  /**
   * Tells the application to watch the module, once bootstraping has completed
   * the modules controller, service, etc. functions will be overwritten to register directly
   * with this application.
   * @param  {[type]} module [description]
   * @return {[type]}        [description]
   */
  app.useModule = function (module) {
    if (preBootModules) {
      preBootModules.push(module);
    } else {
      _.extend(module, register_fns);
    }
    // push it into the apps dependencies
    apps_deps.push(module.name);
    return module;
  };

  app.config(function($locationProvider, $controllerProvider, $compileProvider, $filterProvider, $provide) {
    register_fns.controller = $controllerProvider.register;
    register_fns.directive  = $compileProvider.directive;
    register_fns.factory    = $provide.factory;
    register_fns.service    = $provide.service;
    register_fns.filter     = $filterProvider.register;
  });

  var apps_deps = [
    'grafana.core',
    'ngRoute',
    'ngSanitize',
    '$strap.directives',
    'ang-drag-drop',
    'grafana',
    'pasvaz.bindonce',
    'ui.bootstrap',
    'ui.bootstrap.tpls',
  ];

  var module_types = ['controllers', 'directives', 'factories', 'services', 'filters', 'routes'];

  _.each(module_types, function (type) {
    var module_name = 'grafana.'+type;
    // create the module
    app.useModule(angular.module(module_name, []));
  });

  var preBootRequires = ['app/features/all'];
  var pluginModules = config.bootData.pluginModules || [];

  // add plugin modules
  for (var i = 0; i < pluginModules.length; i++) {
    preBootRequires.push(pluginModules[i]);
  }

  app.boot = function() {
    require(preBootRequires, function () {

      // disable tool tip animation
      $.fn.tooltip.defaults.animation = false;

      // bootstrap the app
      angular
        .element(document)
        .ready(function() {
          angular.bootstrap(document, apps_deps)
            .invoke(['$rootScope', function ($rootScope) {
              _.each(preBootModules, function (module) {
                _.extend(module, register_fns);
              });

              preBootModules = null;

              $rootScope.requireContext = appLevelRequire;
              $rootScope.require = function (deps, fn) {
                var $scope = this;
                $scope.requireContext(deps, function () {
                  var deps = _.toArray(arguments);
                  fn.apply($scope, deps);
                });
              };
            }]);
        });
    });
  };

  return app;
});
