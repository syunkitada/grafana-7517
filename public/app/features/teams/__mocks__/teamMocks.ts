import { Team } from '../../../types';

export const getMultipleMockTeams = (numberOfTeams: number): Team[] => {
  let teams: Team[] = [];
  for (let i = 1; i <= numberOfTeams; i++) {
    teams.push({
      id: i,
      name: `test-${i}`,
      avatarUrl: 'some/url/',
      email: `test-${i}@test.com`,
      memberCount: i,
      search: '',
      members: [],
      groups: [],
    });
  }

  return teams;
};

export const getMockTeam = (): Team => {
  return {
    id: 1,
    name: 'test',
    avatarUrl: 'some/url/',
    email: 'test@test.com',
    memberCount: 1,
    search: '',
    members: [],
    groups: [],
  };
};
