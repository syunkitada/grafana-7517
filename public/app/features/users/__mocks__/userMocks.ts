export const getMockUsers = (amount: number) => {
  const users = [];

  for (let i = 0; i <= amount; i++) {
    users.push({
      avatarUrl: 'url/to/avatar',
      email: `user-${i}@test.com`,
      lastSeenAt: '2018-10-01',
      lastSeenAtAge: '',
      login: `user-${i}`,
      orgId: 1,
      role: 'Admin',
      userId: i,
    });
  }

  return users;
};

export const getMockUser = () => {
  return {
    avatarUrl: 'url/to/avatar',
    email: `user@test.com`,
    lastSeenAt: '2018-10-01',
    lastSeenAtAge: '',
    login: `user`,
    orgId: 1,
    role: 'Admin',
    userId: 2,
  };
};
