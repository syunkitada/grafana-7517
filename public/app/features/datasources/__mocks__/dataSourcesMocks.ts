import { DataSource, DataSourcePermission } from 'app/types';

export const getMockDataSources = (amount: number): DataSource[] => {
  const dataSources = [];

  for (let i = 0; i <= amount; i++) {
    dataSources.push({
      access: '',
      basicAuth: false,
      database: `database-${i}`,
      id: i,
      isDefault: false,
      jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
      name: `dataSource-${i}`,
      orgId: 1,
      password: '',
      readOnly: false,
      type: 'cloudwatch',
      typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
      url: '',
      user: '',
    });
  }

  return dataSources;
};

export const getMockDataSource = (): DataSource => {
  return {
    access: '',
    basicAuth: false,
    basicAuthUser: '',
    basicAuthPassword: '',
    withCredentials: false,
    database: '',
    id: 13,
    isDefault: false,
    jsonData: { authType: 'credentials', defaultRegion: 'eu-west-2' },
    name: 'gdev-cloudwatch',
    orgId: 1,
    password: '',
    readOnly: false,
    type: 'cloudwatch',
    typeLogoUrl: 'public/app/plugins/datasource/cloudwatch/img/amazon-web-services.png',
    url: '',
    user: '',
  };
};

export const getMockDataSourcePermissionsUser = (): DataSourcePermission => {
  return {
    created: '2018-10-10T16:50:45+02:00',
    datasourceId: 1,
    id: 2,
    permission: 1,
    permissionName: 'Query',
    updated: '2018-10-10T16:50:45+02:00',
    userAvatarUrl: '/avatar/926aa85c6bcefa0b4deca3223f337ae1',
    userEmail: 'test@test.com',
    userId: 3,
    userLogin: 'testUser',
  };
};

export const getMockDataSourcePermissionsTeam = (): DataSourcePermission => {
  return {
    created: '2018-10-10T16:57:09+02:00',
    datasourceId: 1,
    id: 6,
    permission: 1,
    permissionName: 'Query',
    team: 'A-team',
    teamAvatarUrl: '/avatar/93c0801b955cbd443a8cfa91a401d7bc',
    teamId: 1,
    updated: '2018-10-10T16:57:09+02:00',
  };
};
