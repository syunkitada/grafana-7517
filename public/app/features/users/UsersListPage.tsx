import React, { PureComponent } from 'react';
import { hot } from 'react-hot-loader';
import { connect } from 'react-redux';
import PageHeader from 'app/core/components/PageHeader/PageHeader';
import UsersActionBar from './UsersActionBar';
import UsersTable from 'app/features/users/UsersTable';
import InviteesTable from './InviteesTable';
import { Invitee, NavModel, OrgUser } from 'app/types';
import appEvents from 'app/core/app_events';
import { loadUsers, loadInvitees, revokeInvite, setUsersSearchQuery, updateUser, removeUser } from './state/actions';
import { getNavModel } from '../../core/selectors/navModel';
import { getInvitees, getUsers, getUsersSearchQuery } from './state/selectors';

export interface Props {
  navModel: NavModel;
  invitees: Invitee[];
  users: OrgUser[];
  searchQuery: string;
  externalUserMngInfo: string;
  loadUsers: typeof loadUsers;
  loadInvitees: typeof loadInvitees;
  setUsersSearchQuery: typeof setUsersSearchQuery;
  updateUser: typeof updateUser;
  removeUser: typeof removeUser;
  revokeInvite: typeof revokeInvite;
}

export interface State {
  showInvites: boolean;
}

export class UsersListPage extends PureComponent<Props, State> {
  state = {
    showInvites: false,
  };

  componentDidMount() {
    this.fetchUsers();
    this.fetchInvitees();
  }

  async fetchUsers() {
    return await this.props.loadUsers();
  }

  async fetchInvitees() {
    return await this.props.loadInvitees();
  }

  onRoleChange = (role, user) => {
    const updatedUser = { ...user, role: role };

    this.props.updateUser(updatedUser);
  };

  onRemoveUser = user => {
    appEvents.emit('confirm-modal', {
      title: 'Delete',
      text: 'Are you sure you want to delete user ' + user.login + '?',
      yesText: 'Delete',
      icon: 'fa-warning',
      onConfirm: () => {
        this.props.removeUser(user.userId);
      },
    });
  };

  onRevokeInvite = code => {
    this.props.revokeInvite(code);
  };

  showInvites = () => {
    this.setState(prevState => ({
      showInvites: !prevState.showInvites,
    }));
  };

  render() {
    const { externalUserMngInfo, invitees, navModel, users } = this.props;

    return (
      <div>
        <PageHeader model={navModel} />
        <div className="page-container page-body">
          <UsersActionBar showInvites={this.showInvites} />
          {externalUserMngInfo && (
            <div className="grafana-info-box">
              <span>{externalUserMngInfo}</span>
            </div>
          )}
          {this.state.showInvites ? (
            <InviteesTable invitees={invitees} revokeInvite={code => this.onRevokeInvite(code)} />
          ) : (
            <UsersTable
              users={users}
              onRoleChange={(role, user) => this.onRoleChange(role, user)}
              onRemoveUser={user => this.onRemoveUser(user)}
            />
          )}
        </div>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    navModel: getNavModel(state.navIndex, 'users'),
    users: getUsers(state.users),
    searchQuery: getUsersSearchQuery(state.users),
    invitees: getInvitees(state.users),
    externalUserMngInfo: state.users.externalUserMngInfo,
  };
}

const mapDispatchToProps = {
  loadUsers,
  loadInvitees,
  setUsersSearchQuery,
  updateUser,
  removeUser,
  revokeInvite,
};

export default hot(module)(connect(mapStateToProps, mapDispatchToProps)(UsersListPage));
