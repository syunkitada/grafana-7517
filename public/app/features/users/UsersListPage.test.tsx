import React from 'react';
import { shallow } from 'enzyme';
import { UsersListPage, Props } from './UsersListPage';
import { NavModel, User } from 'app/types';
import { getMockUser } from './__mocks__/userMocks';
import appEvents from '../../core/app_events';

jest.mock('../../core/app_events', () => ({
  emit: jest.fn(),
}));

const setup = (propOverrides?: object) => {
  const props: Props = {
    navModel: {} as NavModel,
    users: [] as User[],
    searchQuery: '',
    loadUsers: jest.fn(),
    updateUser: jest.fn(),
    removeUser: jest.fn(),
    setUsersSearchQuery: jest.fn(),
  };

  Object.assign(props, propOverrides);

  const wrapper = shallow(<UsersListPage {...props} />);
  const instance = wrapper.instance() as UsersListPage;

  return {
    wrapper,
    instance,
  };
};

describe('Render', () => {
  it('should render component', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Functions', () => {
  it('should emit show remove user modal', () => {
    const { instance } = setup();
    const mockUser = getMockUser();

    instance.onRemoveUser(mockUser);

    expect(appEvents.emit).toHaveBeenCalled();
  });
});
