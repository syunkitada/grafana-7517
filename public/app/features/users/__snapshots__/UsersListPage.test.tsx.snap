// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render List page 1`] = `
<Page
  title="Configuration: Users"
>
  <PageHeader
    model={Object {}}
  />
  <PageContents
    isLoading={false}
  >
    <Connect(UsersActionBar)
      onShowInvites={[Function]}
      showInvites={false}
    />
    <UsersTable
      onRemoveUser={[Function]}
      onRoleChange={[Function]}
      users={Array []}
    />
  </PageContents>
</Page>
`;

exports[`Render should render component 1`] = `
<Page
  title="Configuration: Users"
>
  <PageHeader
    model={Object {}}
  />
  <PageContents
    isLoading={true}
  >
    <Connect(UsersActionBar)
      onShowInvites={[Function]}
      showInvites={false}
    />
  </PageContents>
</Page>
`;
